src/library_stack.js

```diff
@@ -0,0 +1,24 @@
+mergeInto(LibraryManager.library, {
+	emscripten_stack_get_base: function() {
+		return STACK_BASE;
+	},
+	emscripten_stack_get_end: function() {
+		return STACK_MAX;
+	},
+
+#if !WASM_BACKEND
+	// In wasm backend, these functions are implemented
+	// locally in wasm module code as stackSave()/
+	// stackGetFree() functions.
```

Comment:
There they somehow aliased?  How to calls to emscripten_stack_get_current get mapped to stackSave?

.. Ah I saw the getting re-written in binaryen by `FixEmscriptenStackGetWalker`.  This seems a litte complicated.  I wonder if we could have the linker do this for use using `__attribute__(alias)` ?    Maybe in `system/lib/compiler-rt/extras.c` ?   

Out of interest what is you plan for using these in production?  Do they need to be super fast?   If not than just routing them to `stackSave()` in JS would work cleaner and for for both backend and fastcomp.

Reply:
I cannot afford to jump out from Wasm to JS to get the size, the size query should be super-fast. Doing a manual TLS slot when the data is already in a wasm global that is a TLS, would also be extraneous.

Reply:
> I wonder if we could have the linker do this for use using `__attribute__(alias)` ?

I don't think that is currently possible, because LLVM will emit an import of that function in any case - it is not aware that Binaryen will magically introduce the function afterwards, so the missing import will need to be cleaned up anyways.

Reply:
yeah I think you right :(   I'm kind of sad by the extra complexity needed in binaryen for this.   I'd love to find a way to have this all get done by llvm... but I don't have any great ideas right now.

.. maybe we could move the whole stackSave/stackRestore (and now these extra stack functions) into a `.s` assembly that that gets linked as part of compiler-rt?

Reply:
I'll take a look at doing that now for the existing stack functions..

Reply:
It looks like its gonna work!   :)  https://github.com/emscripten-core/emscripten/pull/11166  

I like this soo much better than the binaryen stuff!